gladier/managers/flows_manager.py

```diff
@@ -411,6 +401,26 @@ def run_flow(self, **kwargs):
                 flow = self.flows_client.run_flow(
                     flow_id, self.flow_scope, **kwargs
                 ).data
+            elif gapie.http_status == 400:
+                # Typically, 400s with strange text error messages are due to dependent scopes
+                # changing on the flow between runs, eg add a new Transfer AP and now it needs
+                # a tranfer scope, which the previous access token didn't have. Attempt to pick
+                # the dependent scope problem out of any generic 400, or re-raise if not.
+                try:
+                    log.debug("Encountered error when running flow", exc_info=True)
+                    automate_error_message = json.loads(gapie.text)
+                    detail_message = automate_error_message["error"]["detail"]
+                    if "unable to get tokens for scopes" in detail_message:
```

Comment:
Yeah, it's a bit unfortunate. Ideally we would check for some kind of code, since error messages can easily change. At some point hopefully we will and some of this jank can go away.